daffodil-runtime1/src/main/scala/org/apache/daffodil/processors/parsers/SequenceParserBases.scala

```diff
@@ -261,6 +262,10 @@ abstract class SequenceParserBase(
               // So we mask the failure, and exit the sequence successfully
               pstate.setSuccess()
               isDone = true
+              // If we're masking the failure, we don't want the error dianostics
+              // to flow up. Restore the diagnostics from before the parse
+              // attempt
+              pstate.diagnostics = diagnosticsBeforeAttempt
```

Comment:
This would seem to be needed everywhere we do pstate.setSuccess().  So I did a search for these. These are in fact the only two places where this is needed. :+1: